samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,222 @@
+﻿//--------------------------------------------------------------------------------- 
+// <copyright file="Program.cs" company="Microsoft">
+// Microsoft (R)  Azure SDK 
+// Software Development Kit 
+//  
+// Copyright (c) Microsoft Corporation. All rights reserved.   
+// 
+// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,  
+// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES  
+// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.  
+// </copyright>
+//---------------------------------------------------------------------------------
+
+namespace ChangefeedMigrationSample
+{
+    using DocumentDB.ChangeFeedProcessor;
+    using Microsoft.Azure.Documents;
+    using Microsoft.Azure.Documents.Client;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    // ------------------------------------------------------------------------------------------------
+    // This sample demonstrates using change processor library to read changes from source collection 
+    // to destination collection 
+    // ------------------------------------------------------------------------------------------------
+
+    public class DeviceReading
+    {
+        [JsonProperty("id")]
+        public string Id;
+
+        [JsonProperty("deviceId")]
+        public string DeviceId;
+
+        [JsonConverter(typeof(IsoDateTimeConverter))]
+        [JsonProperty("readingTime")]
+        public DateTime ReadingTime;
+
+        [JsonProperty("metricType")]
+        public string MetricType;
+
+        [JsonProperty("unit")]
+        public string Unit;
+
+        [JsonProperty("metricValue")]
+        public double MetricValue;
+    }
+
+    class DocumentFeedObserver : IChangeFeedObserver
+    {
+        private static int s_totalDocs = 0;
+
+        public Task OpenAsync(ChangeFeedObserverContext context)
+        {
+            Console.WriteLine("Worker opened, {0}", context.PartitionKeyRangeId);
+            return Task.CompletedTask;  // Requires targeting .NET 4.6+.
+        }
+
+        public Task CloseAsync(ChangeFeedObserverContext context, ChangeFeedObserverCloseReason reason)
+        {
+            Console.WriteLine("Worker closed, {0}", context.PartitionKeyRangeId);
+            return Task.CompletedTask;
+        }
+
+        public Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
+        {
+            DocumentCollectionInfo destCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri("httpsURI"),
+                MasterKey = "PrimaryKey",
+                DatabaseName = "DestDBname",
+                CollectionName = "DestCollName"
+            };
+
+            DocumentClient newClient = new DocumentClient(destCollectionLocation.Uri, destCollectionLocation.MasterKey);
```

Comment:
This would try to create the destination collection on every Change Feed change. Move the destination collection creation to the Main method and maintain a private instance of the DocumentClient to reuse on the ProcessChangesAsync. Maintaining a single DocumentClient instead of recreating it continuously is [recommended](https://docs.microsoft.com/en-us/azure/cosmos-db/performance-tips#sdk-usage).